gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/store/ServerGroupUtils.java

```diff
@@ -110,7 +110,7 @@ public static String getMyGroups() {
   /** return true if this VM is member of one of the given groups */
   public static boolean isGroupMember(Set<String> groups) {
     if (groups == null || groups.size() == 0) {
-      return true;
+      return false;
```

Comment:
hmm...true missed it. I think for here instead of using this member directly we can just check whether the group size is 0 or group is null then consider it as a non-lead node.